diff --git a/src/manager.cpp b/src/manager.cpp
index b50d8f9..eac5c7d 100644
--- a/src/manager.cpp
+++ b/src/manager.cpp
@@ -239,4 +239,13 @@ Rfkill *Manager::rfkill() const
     return d->m_rfkill;
 }
 
+PendingCall *Manager::pairWithDevice(const QString &address)
+{
+    BluezQt::DevicePtr device = deviceForAddress(address);
+    if (device) {
+        return device->pair();
+    }
+    return new PendingCall(PendingCall::InternalError, QStringLiteral("Device unknown!"), this);
+}
+
 } // namespace BluezQt
diff --git a/src/manager.h b/src/manager.h
index fbe4292..b28eacf 100644
--- a/src/manager.h
+++ b/src/manager.h
@@ -297,6 +297,16 @@ public:
 
     Rfkill *rfkill() const;
 
+    /**
+     * Pairs with a device.
+     *
+     * Possible errors: PendingCall::DoesNotExist
+     *
+     * @param address address of the device
+     * @return pending call with null if the device is known, otherwise with the device object path
+     */
+    PendingCall *pairWithDevice(const QString &address);
+
 Q_SIGNALS:
     /**
      * Indicates that operational state have changed.
