diff --git a/src/macros.h b/src/macros.h
index 408646f..72c3b09 100644
--- a/src/macros.h
+++ b/src/macros.h
@@ -15,21 +15,27 @@
 #define PROPERTY_CHANGED(var, type_cast, signal) \
     if (var != value.type_cast()) { \
         var = value.type_cast(); \
-        Q_EMIT q.lock()->signal(var); \
+        if (q.lock()) { \
+            Q_EMIT q.lock()->signal(var); \
+        }
     }
 
 #define PROPERTY_CHANGED2(var, value, signal) \
     const auto &val = value; \
     if (var != val) { \
         var = val; \
-        Q_EMIT q.lock()->signal(var); \
+        if (q.lock()) { \
+            Q_EMIT q.lock()->signal(var); \
+        }
     }
 
 // Clears property value and emit propertyChanged signal
 #define PROPERTY_INVALIDATED(var, empty, signal) \
     if (var != empty) { \
         var = empty; \
-        Q_EMIT q.lock()->signal(var); \
+        if (q.lock()) { \
+            Q_EMIT q.lock()->signal(var); \
+        }
     }
 
 #endif // BLUEZQT_MACROS_H
