diff --git a/src/agent.h b/src/agent.h
index 853f8ff..2b534f5 100644
--- a/src/agent.h
+++ b/src/agent.h
@@ -48,6 +48,7 @@ public:
         DisplayYesNo = 1,
         KeyboardOnly = 2,
         NoInputNoOutput = 3,
+        KeyboardDisplay = 4,
     };
     Q_ENUM(Capability)
 
diff --git a/src/manager.cpp b/src/manager.cpp
index c6c97cc..8308e59 100644
--- a/src/manager.cpp
+++ b/src/manager.cpp
@@ -171,6 +171,9 @@ PendingCall *Manager::registerAgent(Agent *agent)
     case Agent::NoInputNoOutput:
         capability = QStringLiteral("NoInputNoOutput");
         break;
+    case Agent::KeyboardDisplay:
+        capability = QStringLiteral("KeyboardDisplay");
+        break;
     default:
         capability = QStringLiteral("DisplayYesNo");
         break;
