diff --git a/src/manager.cpp b/src/manager.cpp
index b50d8f9..5909c86 100644
--- a/src/manager.cpp
+++ b/src/manager.cpp
@@ -230,7 +230,7 @@ PendingCall *Manager::unregisterProfile(Profile *profile)
 #if BLUEZQT_BUILD_DEPRECATED_SINCE(5, 57)
 MediaPtr Manager::media() const
 {
-    return usableAdapter() ? usableAdapter()->media() : nullptr;
+    return usableAdapter() ? usableAdapter()->media() : QSharedPointer<BluezQt::Media>(nullptr);
 }
 #endif
 
